### PR TITLE
change config file logic

### PR DESCRIPTION
**Pull request by dshimo**
addresses *.xml portion of #66
uses server heuristic to support server config files anywhere.
simplifies tracking and diagnostics continue to work

### CODE REVIEW

File: lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java [hunk 1]
```diff
@@ -30,24 +32,30 @@
...
-            return false;
-        }
-
+    public static boolean hasServerRoot(String filePath) {
         XMLInputFactory factory = XMLInputFactory.newInstance();
         XMLEventReader reader = null;
+        File file = null;
+        
         try {
+            file = new File(new URI(filePath).getPath());
```
Comment (cherylking) [inline lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java:41]: just curious why we have to create a URI and get the path from that to create the file? What is the format of the filePath? I know we changed `\` to `/` in the calling method.

**Review by cherylking — COMMENTED** [review]
Just the one question

File: lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java (hunk 1 above)
Comment (dshimo) [reply lemminx-liberty/src/main/java/io/openliberty/tools/langserver/lemminx/util/XmlReader.java:41]: I learned checking `file.exists()` with a `File` constructed with a string in the URI format returns `false`. We get URI formatting from using DOMDocument

File: lemminx-liberty/src/test/java/io/openliberty/LibertyDiagnosticTest.java [hunk 2]
```diff
@@ -141,15 +141,5 @@ public void testDiagnosticsForInclude() throws IOException {
...
-
-        Collection<LibertyWorkspace> workspaceFolders = LibertyProjectsManager.getInstance().getLibertyWorkspaceFolders();
-        assertTrue(workspaceFolders.size() == 1);
-
-        LibertyWorkspace libWorkspace = workspaceFolders.iterator().next();
-
-        assertTrue(libWorkspace.hasConfigFile(new File("src/test/resources/empty_server.xml").getCanonicalPath()));
-        assertFalse(libWorkspace.hasConfigFile("MISSING FILE"));
-        assertFalse(libWorkspace.hasConfigFile("MULTI LINER"));
-        assertFalse(libWorkspace.hasConfigFile("MISSING FILE.xml"));
```
Comment (cherylking) [inline lemminx-liberty/src/test/java/io/openliberty/LibertyDiagnosticTest.java:153]: Should we test that `isConfigXMLFile` works correctly for some of these files?

**Review by cherylking — CHANGES_REQUESTED** [review]

**Review by cherylking — APPROVED** [review]

**Pull request merged**